khal/khalendar/khalendar.py

```diff
@@ -291,36 +113,40 @@ def default_calendar_name(self, default):
         if default is None:
             self._default_calendar_name = default
         elif default not in self.names:
-            raise ValueError('Unknown calendar: {}'.format(default))
+            raise ValueError('Unknown calendar: {0}'.format(default))
 
-        readonly = self._calnames[default].readonly
+        readonly = self._calendars[default].get('readonly', False)
 
         if not readonly:
             self._default_calendar_name = default
         else:
-            raise ValueError('Default calendar is read-only: {}'.format(default))
+            raise ValueError(
+                'Calendar "{0}" is read-only and cannot be used as default'.format(default))
+
+    def _local_ctag(self, calendar):
+        stat = os.stat(self._calendars[calendar]['path'])
+        mtime = getattr(stat, 'st_mtime_ns', None)
+        if mtime is None:
+            mtime = stat.st_mtime
+        import math
+        return str(int(math.floor(mtime * 1e9)))
```

Comment:
True, I was experimenting and not finished here. Also I'm still not sure why the `sleep(1)` is needed.